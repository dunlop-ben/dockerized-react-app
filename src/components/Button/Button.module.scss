.button {
  background-color: #000000;
  border: 2px solid #86bb24;
  border-radius: 3px;
  color: #fff;
}
