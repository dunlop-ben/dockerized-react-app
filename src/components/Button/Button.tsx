// Packages
import React, { FunctionComponent, MouseEvent, ReactChild } from 'react';

import styles from './Button.module.scss';

type Props = {
  children: ReactChild,
  onClick: (e: MouseEvent<HTMLElement>) => void,
};

const Button: FunctionComponent<Props> = ({ children, onClick: handleClick }) => (
  <button className={styles.button} onClick={handleClick} type="button">
    {children}
  </button>
);

export default Button;
