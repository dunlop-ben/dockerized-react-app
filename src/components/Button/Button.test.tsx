// Packages
import React from 'react';
import { shallow } from 'enzyme';

// Components
import Button from './Button';

describe('<Button />', () => {
  let component: any;

  const props = {
    children: 'Button',
    onClick: jest.fn(),
  };

  beforeEach(() => {
    component = shallow(<Button {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('sets onClick correctly', () => {
    component.simulate('click');

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
