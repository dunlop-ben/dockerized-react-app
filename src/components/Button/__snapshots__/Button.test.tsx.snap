// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders without crashing 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
  type="button"
>
  Button
</button>
`;
